### PR TITLE
feat: allow subscriptions to inherit primary model auth rules for relational fields behind a feature flag

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Allow subscriptions to inherit primary model auth rules for relational fields if `subscriptionsInheritPrimaryAuth` is set to `true`.

##### CDK / CloudFormation Parameters Changed
NA

#### Issue #, if available
TODO: To be updated later.

#### Description of how you validated changes
- New E2E tests
- New unit tests
- Manual testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by phani-srikar — APPROVED** [review]

File: packages/amplify-graphql-api-construct/src/types.ts [hunk 1]
```diff
@@ -427,6 +427,13 @@ export interface TranslationBehavior {
    */
   readonly useSubUsernameForDefaultIdentityClaim: boolean;
 
+  /**
+   * When disabled, relational fields on the mutation response are over-redacted to protect the data from subscribed users
+   * to correctly apply the auth rules.
```
Comment (palpatim) [inline packages/amplify-graphql-api-construct/src/types.ts:432]: Suggest something like:

```
   * When enabled, suppresses default behavior of redacting relational fields when auth rules do not exactly match.
   * See <link to doc update>
```

File: packages/amplify-graphql-api-construct/src/types.ts [hunk 2]
```diff
@@ -522,6 +529,13 @@ export interface PartialTranslationBehavior {
    */
   readonly useSubUsernameForDefaultIdentityClaim?: boolean;
 
+  /**
+   * When disabled, relational fields on the mutation response are over-redacted to protect the data from subscribed users
+   * to correctly apply the auth rules.
```
Comment (palpatim) [inline packages/amplify-graphql-api-construct/src/types.ts:534]: Suggest something like:

```
   * When enabled, suppresses default behavior of redacting relational fields when auth rules do not exactly match.
   * See <link to doc update>
```

**Review by palpatim — CHANGES_REQUESTED** [review]
Please adjust doc comments for feature flag and link to published docs explaining the behavior

File: packages/amplify-graphql-api-construct/src/types.ts (hunk 1 above)
Comment (sundersc) [reply packages/amplify-graphql-api-construct/src/types.ts]: Updated the comment. Doc update isn't ready, will update once we have the docs update are published.

File: packages/amplify-graphql-api-construct/src/types.ts (hunk 2 above)
Comment (sundersc) [reply packages/amplify-graphql-api-construct/src/types.ts]: Updated the comment. Doc update isn't ready, will update once we have the docs update are published.

**Review by palpatim — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**